src/SparkTestBase.sol

```diff
@@ -101,34 +217,61 @@ abstract contract SparkTestBase is Test {
 }
 
-abstract contract SparklendTestBase is ProtocolV3TestBase, SparkTestBase {
+/// @dev assertions specific to sparklend, which are not run on chains where
+/// it is not deployed
+abstract contract SparklendTests is ProtocolV3TestBase, SpellRunner {
     using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+    using DomainHelpers for StdChains.Chain;
+    using DomainHelpers for Domain;
 
     bool internal disableExportDiff;
     bool internal disableE2E;
 
+    /// @notice local to market currently under test
     IACLManager                    internal aclManager;
+    /// @notice local to market currently under test
     IPool                          internal pool;
-    IPoolAddressesProvider         internal poolAddressesProvider;
-    IPoolAddressesProviderRegistry internal poolAddressesProviderRegistry;
+    /// @notice local to market currently under test
     IPoolConfigurator              internal poolConfigurator;
+    /// @notice local to market currently under test
     IAaveOracle                    internal priceOracle;
 
+    modifier onChain(ChainId chainId) override virtual {
+        ChainId currentChain = ChainIdUtils.fromUint(block.chainid);
+        chainSpellMetadata[chainId].domain.selectFork();
+        // this mimics the logic of legacy spells where they had a
+        // `loadPoolContext` call in the setup of every test contract involving
+        // sparklend, while not overriding explicit pool context setup if on
+        // nested modifier invocations
+        if(address(pool) == address(0)){
+            loadPoolContext(_getPoolAddressesProviderRegistry().getAddressesProvidersList()[0]);
+        }
+        _;
+        chainSpellMetadata[currentChain].domain.selectFork();
+    }
+
     function loadPoolContext(address poolProvider) internal {
-        poolAddressesProvider = IPoolAddressesProvider(poolProvider);
+        IPoolAddressesProvider poolAddressesProvider = IPoolAddressesProvider(poolProvider);
         pool                  = IPool(poolAddressesProvider.getPool());
         poolConfigurator      = IPoolConfigurator(poolAddressesProvider.getPoolConfigurator());
         aclManager            = IACLManager(poolAddressesProvider.getACLManager());
         priceOracle           = IAaveOracle(poolAddressesProvider.getPriceOracle());
     }
 
-    function testSpellExecutionDiff() public {
-        address[] memory poolProviders = poolAddressesProviderRegistry.getAddressesProvidersList();
-        string memory prefix = string(abi.encodePacked(id, '-', domain));
+    function test_ETHEREUM_SpellExecutionDiff() public {
+        testSpellExecutionDiff(ChainIdUtils.Ethereum());
+    }
+
+    function test_GNOSIS_SpellExecutionDiff() public {
+        vm.skip(chainSpellMetadata[ChainIdUtils.Gnosis()].payload == address(0));
+        testSpellExecutionDiff(ChainIdUtils.Gnosis());
+    }
+
+    function testSpellExecutionDiff(ChainId chainId) onChain(chainId) internal {
```

Comment:
Again we don't want to prefix with `test` if this is not an actual test to be ran.